langdetect/detector.py

```diff
@@ -180,13 +180,17 @@ def _init_probability(self):
 
     def _extract_ngrams(self):
         '''Extract n-grams from target text.'''
+        RANGE = range(1, NGram.N_GRAM + 1)
+
         result = []
         ngram = NGram()
         for ch in self.text:
             ngram.add_char(ch)
-            for n in xrange(1, NGram.N_GRAM+1):
-                w = ngram.get(n)
-                if w is not None and w in self.word_lang_prob_map:
+            if ngram.capitalword:
+                continue
+            for n in RANGE:
+                w = ngram.grams[-n:]
+                if w and w != ' ' and w in self.word_lang_prob_map:
```

Comment:
This condition doesn't mimic the conditions in ``get(n)`` exactly and can produce different results.
For example the part ``len(self.grams) < n`` is not handled and therefore some short ngrams will be inserted to ``result`` multiple times.